Dalamud/Interface/Internal/Windows/Data/Widgets/NetworkMonitorWidget.cs

```diff
@@ -0,0 +1,172 @@
+            this.opCodeDict.TryAdd(code, (name, this.GetSizeFromName(name)));
+    }
+    
+    /// <inheritdoc/>
+    public void Draw()
+    {
+        var network = Service<GameNetwork>.Get();
+        if (ImGui.Checkbox("Track Network Packets", ref this.trackNetwork))
+        {
+            if (this.trackNetwork)
+            {
+                network.NetworkMessage += this.OnNetworkMessage;
+            }
+            else
+            {
+                network.NetworkMessage -= this.OnNetworkMessage;
+            }
+        }
+
+        ImGui.SetNextItemWidth(ImGui.GetContentRegionAvail().X / 2);
+        if (ImGui.DragInt("Stored Number of Packets", ref this.trackedPackets, 0.1f, 1, 512))
+        {
+            this.trackedPackets = Math.Clamp(this.trackedPackets, 1, 512);
+        }
+
+        this.DrawFilterInput();
+
+        ImGuiTable.DrawTable(string.Empty, this.packets, this.DrawNetworkPacket, ImGuiTableFlags.SizingFixedFit | ImGuiTableFlags.RowBg, "Direction", "OpCode", "Source", "Target", "Data");
+    }
+
+    private void DrawNetworkPacket(NetworkPacketData data)
+    {
+        ImGui.TableNextColumn();
+        ImGui.TextUnformatted(data.Direction.ToString());
+
+        ImGui.TableNextColumn();
+        ImGui.TextUnformatted(this.opCodeDict.TryGetValue(data.OpCode, out var pair) ? pair.Item1 : data.OpCode.ToString());
+
+        ImGui.TableNextColumn();
+        ImGui.TextUnformatted($"0x{data.SourceActorId:X}");
+
+        ImGui.TableNextColumn();
+        ImGui.TextUnformatted($"0x{data.TargetActorId:X}");
+
+        ImGui.TableNextColumn();
+        if (data.Data.Count > 0)
+        {
+            ImGui.TextUnformatted(string.Join(" ", data.Data.Select(b => b.ToString("X2"))));
+        }
+    }
+
+    private void DrawFilterInput()
+    {
+        var invalidRegEx = this.filterString.Length > 0 && this.trackedOpCodes == null;
+        using var style = ImRaii.PushStyle(ImGuiStyleVar.FrameBorderSize, 2 * ImGuiHelpers.GlobalScale, invalidRegEx);
+        using var color = ImRaii.PushColor(ImGuiCol.Border, 0xFF0000FF, invalidRegEx);
+        ImGui.SetNextItemWidth(ImGui.GetContentRegionAvail().X);
+        if (!ImGui.InputTextWithHint("##Filter", "Regex Filter OpCodes...", ref this.filterString, 256))
+        {
+            return;
+        }
+
+        if (this.filterString.Length == 0)
+        {
+            this.trackedOpCodes = null;
+        }
+        else
+        {
+            try
+            {
+                this.trackedOpCodes = new Regex(this.filterString, RegexOptions.Compiled | RegexOptions.ExplicitCapture);
+            }
+            catch
+            {
+                this.trackedOpCodes = null;
+            }
+        }
+    }
+
+    private void OnNetworkMessage(nint dataPtr, ushort opCode, uint sourceActorId, uint targetActorId, NetworkMessageDirection direction)
+    {
+        if (this.trackedOpCodes == null || this.trackedOpCodes.IsMatch(this.OpCodeToString(opCode)))
```

Comment:
Can we get an option to invert this match?
Its simple enough to write a regex to show 790 or 709 `(790|709)` but it becomes a lot more complex to *hide* 790 and 709.
Maybe there's a simple way to write the regex for this.
Often I don't know what opcode I am looking for and just want to trigger the event I need, and then narrow down the opcodes that I know I don't want.

Reply:
Negative Lookahead is a thing in Regex with `(?!<clause>)` this meaning `^(?!(790|709))` should hide 709 and 790

Reply:
> Negative Lookahead is a thing in Regex with `(?!<clause>)` this meaning `^(?!(790|709))` should hide 709 and 790

![image](https://github.com/goatcorp/Dalamud/assets/9083275/d2019317-f244-481a-9881-1e5458504ba5)

That does not work.

Reply:
![proof](https://screenshots.wildwolf.dev/WindowsTerminal_b2a8635b07.png)

C# console says that it works the same way that regex101 shows it also matching the correct match line